app/src/main/java/com/auth0/app/MyActivity.java

```diff
@@ -58,7 +58,9 @@ public void onClick(View v) {
         emailButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                startActivity(new Intent(MyActivity.this, LockEmailActivity.class));
+                Intent intent = new Intent(MyActivity.this, LockEmailActivity.class);
```

Comment:
Lets add the following static method to `LockEmailActivity`:

```java
public static void showFrom(Activity activity, boolean useMagicLink) {
    Intent intent = new Intent(activity, LockEmailActivity.class);
    intent.putExtra(LockEmailActivity.USE_MAGIC_LINK, useMagicLink);
    activity.startActivity(intent);
}
```

Then we can just show it like this

```java
LockEmailActivity.showFrom(this, true);
```